internal/server/server.go

```diff
@@ -32,3 +41,70 @@ func New(logger *log.Logger, client Client, templates map[string]*template.Templ
 
 	return http.StripPrefix(prefix, mux)
 }
+
+type RedirectError string
+
+func (e RedirectError) Error() string {
+	return "redirect to " + string(e)
+}
+
+func (e RedirectError) To() string {
+	return string(e)
+}
+
+type StatusError int
+
+func (e StatusError) Error() string {
+	code := e.Code()
+
+	return fmt.Sprintf("%d %s", code, http.StatusText(code))
+}
+
+func (e StatusError) Code() int {
+	return int(e)
+}
+
+type Handler func(w http.ResponseWriter, r *http.Request) error
+
+func errorHandler(name string, logger *log.Logger, tmpl403 Template, prefix, siriusURL string, next Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if err := next(w, r); err != nil {
+			if status, ok := err.(StatusError); ok {
+				if status.Code() == http.StatusForbidden {
+					type v struct {
+						SiriusURL string
+						Path      string
+					}
+
+					w.WriteHeader(http.StatusForbidden)
+					err = tmpl403.ExecuteTemplate(w, "page", v{
+						SiriusURL: siriusURL,
+						Path:      "",
+					})
+
+					if err == nil {
+						return
+					}
+
+					logger.Printf("%s: %v\n", name, err)
+				}
+
+				http.Error(w, "", status.Code())
+				return
+			}
+
+			if err == sirius.ErrUnauthorized {
+				http.Redirect(w, r, siriusURL+"/auth", http.StatusFound)
```

Comment:
Sorry for the scope creep, but we will need to use a different `siriusURL` for this as it will end up being some random internal AWS URL that gets injected.